### PR TITLE
Remove "atLeast" from the internal query APIs

### PR DESCRIPTION
**Pull request by jsteemann**
Removes `atLeast` from the `atLeast, atMost` internal query APIs

### CODE REVIEW

File: arangod/Aql/ClusterBlocks.cpp [hunk 1]
```diff
@@ -228,7 +228,7 @@ bool GatherBlock::hasMore() {
     for (size_t i = 0; i < _gatherBlockBuffer.size(); i++) {
       if (!_gatherBlockBuffer.at(i).empty()) {
         return true;
-      } else if (getBlock(i, DefaultBatchSize(), DefaultBatchSize())) {
```
Comment (graetzer) [inline arangod/Aql/ClusterBlocks.cpp]: why was this not 1 in the first place?

File: arangod/Aql/ClusterBlocks.cpp [hunk 2]
```diff
@@ -1019,9 +1015,9 @@ bool DistributeBlock::getBlockForClient(size_t atLeast, size_t atMost,
   std::vector<std::deque<std::pair<size_t, size_t>>>& buf = _distBuffer;
   // it should be the case that buf.at(clientId) is empty
 
-  while (buf.at(clientId).size() < atLeast) {
+  while (buf.at(clientId).size() < atMost) {
```
Comment (graetzer) [inline arangod/Aql/ClusterBlocks.cpp]: lol

Comment (jsteemann) [reply arangod/Aql/ClusterBlocks.cpp]: I think it made sense with `atLeast`/`atMost`.

File: arangod/Aql/ClusterBlocks.cpp (hunk 1 above)
Comment (jsteemann) [reply arangod/Aql/ClusterBlocks.cpp]: good question...

File: arangod/Aql/ExecutionBlock.cpp [hunk 3]
```diff
@@ -397,15 +393,15 @@ int ExecutionBlock::getOrSkipSome(size_t atLeast, size_t atMost, bool skipping,
   // if _buffer.size() is > 0 then _pos points to a valid place . . .
   BlockCollector collector(&_engine->_itemBlockManager);
 
-  while (skipped < atLeast) {
+  while (skipped < atMost) {
     if (_buffer.empty()) {
       if (skipping) {
         size_t numActuallySkipped = 0;
-        _dependencies[0]->skip(atLeast - skipped, numActuallySkipped);
-        skipped = atLeast;
```
Comment (graetzer) [inline arangod/Aql/ExecutionBlock.cpp]: will this change behaviour of queries where batchSize is not 1000 ? i.e. Streaming querues

File: arangod/Aql/ExecutionBlock.cpp [hunk 4]
```diff
@@ -192,14 +191,14 @@ void ExecutionBlock::traceGetSomeEnd(AqlItemBlock const* result) const {
 
 /// @brief getSome, gets some more items, semantic is as follows: not
 /// more than atMost items may be delivered. The method tries to
-/// return a block of at least atLeast items, however, it may return
+/// return a block of at atMost items, however, it may return
```
Comment (mchacki) [inline arangod/Aql/ExecutionBlock.cpp]: at most atMost items

File: arangod/Aql/ExecutionBlock.cpp [hunk 5]
```diff
@@ -300,14 +299,12 @@ bool ExecutionBlock::getBlock(size_t atLeast, size_t atMost) {
 /// the idea is that somebody who wants to call the generic functionality
 /// in a derived class but wants to modify the results before the register
 /// cleanup can use this method, internal use only
-AqlItemBlock* ExecutionBlock::getSomeWithoutRegisterClearout(size_t atLeast,
-                                                             size_t atMost) {
+AqlItemBlock* ExecutionBlock::getSomeWithoutRegisterClearout(size_t atMost) {
   DEBUG_BEGIN_BLOCK();
-  TRI_ASSERT(0 < atLeast && atLeast <= atMost);
```
Comment (mchacki) [inline arangod/Aql/ExecutionBlock.cpp]: We loose the indirect assert (0 < atMost) here

File: arangod/Aql/ExecutionBlock.cpp [hunk 6]
```diff
@@ -397,15 +393,15 @@ int ExecutionBlock::getOrSkipSome(size_t atLeast, size_t atMost, bool skipping,
...
 
-  while (skipped < atLeast) {
+  while (skipped < atMost) {
     if (_buffer.empty()) {
       if (skipping) {
         size_t numActuallySkipped = 0;
-        _dependencies[0]->skip(atLeast - skipped, numActuallySkipped);
-        skipped = atLeast;
+        _dependencies[0]->skip(atMost - skipped, numActuallySkipped);
+        skipped = atMost;
```
Comment (mchacki) [inline arangod/Aql/ExecutionBlock.cpp]: i think this assignment is not correct. (it was not in original either...)
Shouldn't it be:
skipped += numActuallySkipped

File: arangod/Aql/ExecutionBlock.h [hunk 7]
```diff
@@ -146,16 +146,16 @@ class ExecutionBlock {
   virtual int shutdown(int);
 
   /// @brief getOne, gets one more item
-  virtual AqlItemBlock* getOne() { return getSome(1, 1); }
+  virtual AqlItemBlock* getOne() { return getSome(1); }
 
   /// @brief getSome, gets some more items, semantic is as follows: not
   /// more than atMost items may be delivered. The method tries to
-  /// return a block of at least atLeast items, however, it may return
+  /// return a block of at atMost items, however, it may return
```
Comment (mchacki) [inline arangod/Aql/ExecutionBlock.h]: at most atMost

File: arangod/Aql/ExecutionBlock.h [hunk 8]
```diff
@@ -191,10 +191,10 @@ class ExecutionBlock {
  public:
   /// @brief getSome, skips some more items, semantic is as follows: not
   /// more than atMost items may be skipped. The method tries to
-  /// skip a block of at least atLeast items, however, it may skip
+  /// skip a block of at atMost items, however, it may skip
```
Comment (mchacki) [inline arangod/Aql/ExecutionBlock.h]: at most atMost

**Pull request merged**